src/main/java/lotto/model/LottoRank.java

```diff
@@ -25,10 +27,15 @@ public int getCashPrize() {
         return cashPrize;
     }
 
-    public static LottoRank findByHits(int hits) {
-        return Arrays.stream(LottoRank.values())
+    public static LottoRank findByHits(int hits, boolean hasBonus) {
+        return filterBonusLottoRank(hasBonus)
                 .filter(lottoRank -> lottoRank.hits == hits)
                 .findAny()
                 .orElse(MISS);
```

Comment:
enum 에서 보너스번호 일치여부도 관리하면 가독성 차원에서도 좋고 아래와 같이 표현할 수도 있어요.
```suggestion
        return Arrays.stream(LottoRank.values())
                   .filter(it -> it.hits  == hits)
                   .filter(it -> it.bonus == hasBonus)
                   .findAny()
                   .orElse(MISS);
```

Reply:
흐름상 가독성도 좋고 불필요한 메소드도 필요 없겠네요 ! 
조언 감사드립니다  👍🏻👍🏻👍🏻